Pass environment variables to falcoctl install and follow

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks! -->

**Describe the bug**

Hello, I am trying to pass some environment variables to the falcoctl `init` and `follow` containers.

I noticed the expected parameters for the env key is a [map](https://github.com/falcosecurity/charts/blob/master/falco/values.yaml#L349), but I am not 100% sure how to pass my env variables to this map. In fact I was expecting a list.

If I try to pass the variables in this way:

```yaml
falcoctl:
  artifact:
    install:
      enabled: true
      env:
        envVariable1:
          valueFrom:
            secretKeyRef:
              name: secret
              key: password
        envVariable2: value
```

The installation of course fails with this error, as the env: field in the pod manifest is expecting a list of env:

```
Error: INSTALLATION FAILED: DaemonSet in version "v1" cannot be handled as a DaemonSet: json: cannot unmarshal object into Go struct field Container.spec.template.spec.containers.env of type []v1.EnvVar
```

If I try to force-pass an array

```yaml
falcoctl:
  artifact:
    install:
      enabled: true
      env:
        - name: envVariable1
          valueFrom:
            secretKeyRef:
              name: secret
              key: password
        - name: envVariable2
          value: value
```

the installation seems to succeed but I get these warning messages:

```
coalesce.go:220: warning: cannot overwrite table with non table for dragonfly-data-collector.falco.falcoctl.artifact.install.env (map[])
coalesce.go:220: warning: cannot overwrite table with non table for dragonfly-data-collector.falco.falcoctl.artifact.install.env (map[])
```

**How to reproduce it**

With this values.yaml:

```
falcoctl:
  artifact:
    install:
      enabled: true
      env:
        envVariable1:
          valueFrom:
            secretKeyRef:
              name: secret
              key: password
        envVariable2: value
```

The installation fails:

```
helm install falco-test falcosecurity/falco -f ./values.yaml
Error: INSTALLATION FAILED: DaemonSet in version "v1" cannot be handled as a DaemonSet: json: cannot unmarshal object into Go struct field Container.spec.template.spec.initContainers.env of type []v1.EnvVar
```

With this values.yaml

```
falcoctl:
  artifact:
    install:
      enabled: true
      env:
        - name: envVariable1
          valueFrom:
            secretKeyRef:
              name: secret
              key: password
        - name: envVariable2
          value: value
```

Installation succeeds but I get few warning messages

```
helm install falco-test falcosecurity/falco -f ./values.yaml
coalesce.go:220: warning: cannot overwrite table with non table for falco.falcoctl.artifact.install.env (map[])
coalesce.go:220: warning: cannot overwrite table with non table for falco.falcoctl.artifact.install.env (map[])
NAME: falco-test
LAST DEPLOYED: Fri May  5 10:35:54 2023
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Falco agents are spinning up on each node in your cluster. After a few
seconds, they are going to start monitoring your containers looking for
security issues.


No further action should be required.


Tip:
You can easily forward Falco events to Slack, Kafka, AWS Lambda and more with falcosidekick.
Full list of outputs: https://github.com/falcosecurity/charts/tree/master/falcosidekick.
You can enable its deployment with `--set falcosidekick.enabled=true` or in your values.yaml.
See: https://github.com/falcosecurity/charts/blob/master/falcosidekick/values.yaml for configuration values.
```

**Expected behaviour**

I am not sure if this is the expected behavior or I am passing the environment variables in the wrong way.

**Screenshots**


<!-- If applicable, add screenshots to help explain your problem. -->

**Environment**

<!-- Please complete the following info. -->

- Chart version: 3.1.4
- Falco version: 0.34.1
- System info: k8s cluster

**Additional context**

<!-- Add any other context about the problem here. -->

--- a/falco/CHANGELOG.md
+++ b/falco/CHANGELOG.md
@@ -3,6 +3,10 @@
 This file documents all notable changes to Falco Helm Chart. The release
 numbering uses [semantic versioning](http://semver.org).
 
+## v3.1.5
+
+* Use list as default for env parameter of init and follow containers
+
 ## v3.1.4
 
 * Fix typo in values-k8audit file

--- a/falco/Chart.yaml
+++ b/falco/Chart.yaml
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: falco
-version: 3.1.4
+version: 3.1.5
 appVersion: 0.34.1
 description: Falco
 keywords:

--- a/falco/values.yaml
+++ b/falco/values.yaml
@@ -346,7 +346,7 @@ falcoctl:
     install:
       enabled: true
       # -- Extra environment variables that will be pass onto falcoctl-artifact-install init container.
-      env: {}
+      env: []
       # -- Arguments to pass to the falcoctl-artifact-install init container.
       args: ["--verbose"]
       # -- Resources requests and limits for the falcoctl-artifact-install init container.
@@ -361,7 +361,7 @@ falcoctl:
     follow:
       enabled: true
       # -- Extra environment variables that will be pass onto falcoctl-artifact-follow sidecar container.
-      env: {}
+      env: []
       # -- Arguments to pass to the falcoctl-artifact-follow sidecar container.
       args: ["--verbose"]
       # -- Resources requests and limits for the falcoctl-artifact-follow sidecar container.

